packages/react/navigation-menu/src/NavigationMenu.tsx

```diff
@@ -504,13 +506,22 @@ const NavigationMenuLink = React.forwardRef<NavigationMenuLinkElement, Navigatio
           aria-current={active ? 'page' : undefined}
           {...linkProps}
           ref={forwardedRef}
-          onClick={composeEventHandlers(props.onClick, (event) => {
-            const rootContentDismissEvent = new Event(ROOT_CONTENT_DISMISS, {
-              bubbles: true,
-              cancelable: true,
-            });
-            event.target.dispatchEvent(rootContentDismissEvent);
-          })}
+          onClick={(event) => {
```

Comment:
Why did we move away from `composeEventHandlers` here?

Reply:
Because I forgot we had the `checkForDefaultPrevented` flag 😄

Reply:
We really need to refactor to these 2 methods at some point 😀